arm/Microsoft.Web/sites/config-appsettings/deploy.bicep

```diff
@@ -0,0 +1,96 @@
+// ================ //
+// Parameters       //
+// ================ //
+@description('Required. Name of the site parent resource.')
+param appName string
+
+@description('Required. Type of site to deploy.')
+@allowed([
+  'functionapp'
+  'functionapp,linux'
+  'app'
+])
+param kind string
+
+@description('Optional. Required if app of kind functionapp. Resource ID of the storage account to manage triggers and logging function executions.')
+param storageAccountId string = ''
+
+@description('Optional. Resource ID of the app insight to leverage for this resource.')
+param appInsightId string = ''
+
+@description('Optional. For function apps. If true the app settings "AzureWebJobsDashboard" will be set. If false not. In case you use Application Insights it can make sense to not set it for performance reasons.')
+param setAzureWebJobsDashboard bool = contains(kind, 'functionapp') ? true : false
+
+@description('Optional. The app settings key value pairs except for AzureWebJobsStorage, AzureWebJobsDashboard, APPINSIGHTS_INSTRUMENTATIONKEY and APPLICATIONINSIGHTS_CONNECTION_STRING.')
+param appSettingsKeyValuePairs object = {}
+
+@description('Optional. Enable telemetry via the Customer Usage Attribution ID (GUID).')
+param enableDefaultTelemetry bool = true
+
+// =========== //
+// Variables   //
+// =========== //
+var azureWebJobsValues = !empty(storageAccountId) && setAzureWebJobsDashboard == true ? {
```

Comment:
Isn't it easier like this?

```bicep
var azureWebJobsValues = !empty(storageAccountId) ? union({
  AzureWebJobsStorage: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};'
}, ((setAzureWebJobsDashboard == false) ? {
  AzureWebJobsDashboard: 'DefaultEndpointsProtocol=https;AccountName=${storageAccount.name};AccountKey=${storageAccount.listKeys().keys[0].value};'
} : {})) : {} 
```

Reply:
@MrMCake Thanks for the suggestion. I updated it.